src/bosh_azure_cpi/lib/cloud/azure/cloud.rb

```diff
@@ -411,7 +421,7 @@ def resize_disk(disk_cid, new_size)
     def has_disk?(disk_cid)
       with_thread_name("has_disk?(#{disk_cid})") do
         @telemetry_manager.monitor('has_disk?', id: disk_cid) do
-          disk_id = DiskId.parse(disk_cid, _azure_config.resource_group_name)
+          disk_id = CloudIdParser.parse(disk_cid, _azure_config.resource_group_name)
           if disk_id.disk_name.start_with?(MANAGED_DATA_DISK_PREFIX)
             return @disk_manager2.has_data_disk?(disk_id)
           else
```

Comment:
Several CPI methods are only used by cloudcheck. They may not be covered when you deploy CF.
You may want to run `bosh cck` to check whether these methods work.